common/ferrostar/src/algorithms.rs

```diff
@@ -66,8 +49,8 @@ pub fn index_of_closest_segment_origin(location: UserLocation, line: &LineString
         // Find the line segment closest to the user's location
         .min_by(|(_, line1), (_, line2)| {
             // Note: lines don't implement haversine distances
```

Comment:
I think you might want https://docs.rs/geo/latest/geo/algorithm/cross_track_distance/trait.CrossTrackDistance.html

I think geo should one day export that as `Haversine::distance(line, point)`. Relatedly, see https://github.com/georust/geo/issues/1128

Reply:
That... is exactly what we want and I just didn't know what it was named 😅

Reply:
Upon further inspection, this sounds like a great idea, but what I really wanted is highly specific to the [Line](https://docs.rs/geo-types/0.7.13/geo_types/geometry/struct.Line.html) implementation. Perhaps `Line` is actually mis-named, as it is really describing a line _segment_. Consider the case of a perfect NW line `[(0,0), (1,1), ...]`. The algorithm using cross track distance uses the mathematical definition of a line as infinite and passing between the points. This means that even at point (100, 100), the algorithm as specified will return index zero, or point (0, 0)!!

That is not at all what we expect. What we actually want is not the closest abstract line that matches, but the _segment_ nearest to us. I'll add some clarifying comments to the code here.

Reply:
You are better at reading documentation than me! I mistakenly thought that CrossTrackDistance was a measure to a segment.